lib/ws_proxy.rb

```diff
@@ -58,8 +58,7 @@ def start_proxy
       end
     end
     @proxy_port = port
-
-    { :host => host, :port => host_port, :password => password, :proxy_port => proxy_port }
+    { :port => proxy_port, :password => password, :encrypt => @encrypt }
```

Comment:
Am I missing the assignment to `@encrypt`? I can't find it in the whole class.

Reply:
check this file: /foreman/app/controllers/concerns/foreman/controller/console_common.rb
```
module Foreman::Controller::ConsoleCommon
  def console
    @encrypt = Setting[:websockets_encrypt]
    render case @console[:type]
             when 'spice'
               'hosts/console/spice'
             when 'vnc'
               'hosts/console/vnc'
             when 'vmrc'
               'hosts/console/vmrc'
             else
               'hosts/console/log'
           end
  end
end
```